test/ParserTests.hs

```diff
@@ -628,13 +634,30 @@ testIdentifiersMustBeLower = TestCase (
   False
   (isRight $ parseAll (many decl) "" "F:Int\nF=5\nF2:Int->Int\nF2(x)=x"))
 
--- | Inputs should be okay where a normal Input would be
 testNestedExprInWhileOkay :: Test
 testNestedExprInWhileOkay = TestCase (
   assertEqual "Test that unparenthesized nested expressions are allowed in while"
   True
   (isRight $ parseAll expr "" "while x < 10 do x + 1"))
 
+testWellFormedLiteral :: Test
+testWellFormedLiteral = TestCase (
+  assertEqual "Test that well-formed literals parse"
+  True $
+  checkAllParsePass literal lits)
+     where
+        lits = ["1", " 1", "1 ", "True", "False", "-1", "+1", "A", "(((40, 2), Nested, Tuple), 0)",
```

Comment:
+1 is a unary plus, just like -1 is a unary minus. Did you know that `1 + + 1` ==> `1 + (+1)` is a valid BoGL program? Perhaps it could be removed, but that requires messing around with the lexer.

Reply:
Honestly I had no clue, a little bit surprised actually. We can bring this up with Martin, but this seems like a very small concern for the moment.